dipy/reconst/ivim.py

```diff
@@ -0,0 +1,478 @@
+               Radiology 265.3 (2012): 874-881.
+        """
+        if not np.any(gtab.b0s_mask):
+            e_s = "No measured signal at bvalue == 0."
+            e_s += "The IVIM model requires signal measured at 0 bvalue"
+            raise ValueError(e_s)
+
+        ReconstModel.__init__(self, gtab)
+        self.split_b_D = split_b_D
+        self.split_b_S0 = split_b_S0
+        self.bounds = bounds
+        self.tol = tol
+        self.options = options
+        self.scaling = scaling
+
+        if SCIPY_LESS_0_17 and self.bounds is not None:
+            e_s = "Scipy versions less than 0.17 do not support "
+            e_s += "bounds. Please update to Scipy 0.17 to use bounds"
+            raise ValueError(e_s)
+        else:
+            self.bounds = ((0., 0., 0., 0.), (np.inf, 1., 1., 1.))
+
+    @multi_voxel_fit
+    def fit(self, data, mask=None):
+        """ Fit method of the Ivim model class.
+
+        The fitting takes place in the following steps: Linear fitting for D
+        (bvals > 200) and store S0_prime. Another linear fit for S0 (bvals <
+        200).Estimate f using 1 - S0_prime/S0. Use least squares to fit D_star
+        and f.
+
+        We do a final fitting of all four parameters after scaling the
+        parameters and select the set of parameters which make sense physically.
+        The criteria for selecting a particular set of parameters is checking
+        the perfusion fraction. If the fraction is more than 30%, we will reject
+        the solution obtained without scaling.
+
+
+        Parameters
+        ----------
+        data : array
+            The measured signal from one voxel. A multi voxel decorator
+            will be applied to this fit method to scale it and apply it
+            to multiple voxels.
+
+        mask : array
+            A boolean array used to mark the coordinates in the data that
+            should be analyzed that has the shape data.shape[:-1]
+
+        Returns
+        -------
+        IvimFit object
+        """
+        # Use the function estimate_S0_prime_D to get S0_prime and D.
+        S0_prime, D = self.estimate_S0_prime_D(data)
+        S0, D_star_prime = self.estimate_S0_D_star_prime(data)
+        f_guess = 1 - S0_prime / S0
+        x0_guess = np.array([S0, f_guess, D_star_prime, D])
+        try:
+            f, D_star = self.estimate_f_D_star(data, x0_guess)
+        except:
+            warningMsg = "x0 obtained from linear fitting is unfeasibile."
+            warningMsg += "Using parameters from the linear fitting"
+            warnings.warn(warningMsg, UserWarning)
+            f, D_star = f_guess, D_star_prime
+
+        x0 = np.array([S0, f, D_star, D])
+        # Fit parameters again with scaling
+        params_with_scaling = self._leastsq(data, x0, scaling=self.scaling)
+        # Fit parameters without scaling
+        params_no_scaling = self._leastsq(data, x0)
+
+        params_in_mask = np.where(params_no_scaling[1] <= .5,
```

Comment:
Is this double fitting useful? Are there times when no scaling works better than scaling? It just seems like fitting twice would be good to avoid if possible.